File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/cpp_solutions/1351-count-negative-numbers-in-a-sorted-matrix.cpp
#include <vector>

class Solution {
public:
    int countNegatives(std::vector<std::vector<int>>& grid) {
        int m = grid.size();
        int n = grid[0].size();

        int count = 0;
        int row = m - 1; 
        int col = 0;     

        while (row >= 0 && col < n) {
            if (grid[row][col] < 0) {
                count += (n - col);
                row--;
            } else {
                col++;
            }
        }

        return count;
    }
};

File: leetcode/AIcoding/4.GPT-4o/go_solutions/1351-count-negative-numbers-in-a-sorted-matrix.go
func countNegatives(grid [][]int) int {
    m, n := len(grid), len(grid[0])
    count := 0
    for i := 0; i < m; i++ {
        for j := n - 1; j >= 0; j-- {
            if grid[i][j] < 0 {
                count++
            } else {
                break
            }
        }
    }
    return count
}